Rocket.Chat/Views/Chat/New Chat/Cells/DateSeparatorCell.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  DateSeparatorCell.swift
+//  Rocket.Chat
+//
+//  Created by Rafael Streit on 26/09/18.
+//  Copyright © 2018 Rocket.Chat. All rights reserved.
+//
+
+import UIKit
+import RocketChatViewController
+
+final class DateSeparatorCell: UICollectionViewCell, ChatCell {
+    static let identifier = String(describing: DateSeparatorCell.self)
+
+    static let sizingCell: DateSeparatorCell = {
+        guard let cell = DateSeparatorCell.instantiateFromNib() else {
+            return DateSeparatorCell()
+        }
+
+        return cell
+    }()
+
+    @IBOutlet weak var date: UILabel!
+
+    var viewModel: AnyChatItem?
+    var contentViewWidthConstraint: NSLayoutConstraint!
+
+    override func awakeFromNib() {
+        super.awakeFromNib()
+
+        contentView.translatesAutoresizingMaskIntoConstraints = false
+        contentViewWidthConstraint = contentView.widthAnchor.constraint(equalToConstant: UIScreen.main.bounds.width)
+        contentViewWidthConstraint.isActive = true
+    }
+
+    func configure() {
+        guard let viewModel = viewModel?.base as? DateSeparatorChatItem else {
+            return
+        }
+
+        let calendar = Calendar.current
```

Comment:
We should move any code about processing the model and making it ready to present to the `ChatItem` which is serving as a `viewModel` for us.

Reply:
✅